### PR TITLE
Alexey - WI 84: GitHub webhook to ADO publishing flow

### PR DESCRIPTION
**Pull request by alex-ko-dev**

### CODE REVIEW

File: src/test/java/com/checkmarx/flow/cucumber/integration/cxconfig/CxConfigSteps.java [hunk 1]
```diff
@@ -71,19 +72,22 @@
     private ResultsService resultsService;
     private Boolean isPullRequestApproved;
 
-    private FlowService flowService;
+    private final FlowService flowService;
     private String branch;
     private ScanRequest request;
+    private final JiraProperties jiraProperties;
 
     public CxConfigSteps( FlowProperties flowProperties, GitHubService gitHubService,
```
Comment (NimrodGolan) [inline src/test/java/com/checkmarx/flow/cucumber/integration/cxconfig/CxConfigSteps.java:80]: Question - why init all properties via the constructor and not by using the @Autowired annotation?
Is Spring initiate it?

File: src/test/java/com/checkmarx/flow/cucumber/component/webhook/WebHookSteps.java [hunk 2]
```diff
@@ -1,71 +1,59 @@
...
+    @Before("@WebHook")
     public void loadProperties() throws IOException {
         testProperties = TestUtils.getPropertiesFromResource("cucumber/features/componentTests/webhook.properties");
     }
 
     @Given("CxFlow is running as a service")
     public void runAsService() {
-        ConfigurableApplicationContext appContext = TestUtils.runCxFlowAsService();
-        cxFlowPort = appContext.getEnvironment().getProperty("server.port");
+        cxFlowPort = TestUtils.runCxFlowAsService();
```
[the remark this answers is not in the document]
Comment (alex-ko-dev) [reply src/test/java/com/checkmarx/flow/cucumber/component/webhook/WebHookSteps.java:49]: I've checked and it turns out that we do need to run CxFlow explicitly. Of course, this is for the tests that require an actual web app running on a specific port and accepting HTTP requests. 
For tests that just need to inject beans, runCxFlowAsService is not required.

File: src/test/java/com/checkmarx/flow/cucumber/integration/azure/publishing/AzureDevopsClient.java [hunk 3]
```diff
@@ -33,9 +31,8 @@
     // Affects the list of valid issue states. CxFlow assumes this template by default.
```
[the remark this answers is not in the document]
Comment (alex-ko-dev) [reply src/test/java/com/checkmarx/flow/cucumber/integration/azure/publishing/AzureDevopsClient.java:31]: Fixed.

File: src/test/java/com/checkmarx/flow/cucumber/integration/cxconfig/CxConfigSteps.java (hunk 1 above)
Comment (alex-ko-dev) [reply src/test/java/com/checkmarx/flow/cucumber/integration/cxconfig/CxConfigSteps.java:80]: This is actually Orly's code, but I also prefer constructor injection.
Even the Spring team itself "generally advocates constructor injection" (https://docs.spring.io/spring-framework/docs/current/spring-framework-reference/core.html#beans-setter-injection)

File: src/test/java/com/checkmarx/flow/cucumber/integration/azure/publishing/AzureDevopsClient.java [hunk 4]
```diff
@@ -116,19 +115,23 @@ public void updateIssueState(Issue issue, String newState) {
...
+    public int getIssueCount() throws IOException {
+        return getProjectIssueIds().size();
+    }
+
+    private void deleteIssue(String issueId) {
         log.info("Deleting ADO issue, ID: {}", issueId);
-        String url = getIssueDeletionUrl(projectName, issueId);
+        String url = getIssueDeletionUrl(issueId);
         HttpEntity<?> request = getRequestEntity(null);
         restClient.exchange(url, HttpMethod.DELETE, request, String.class);
```
[the remark this answers is not in the document]
Comment (alex-ko-dev) [reply src/test/java/com/checkmarx/flow/cucumber/integration/azure/publishing/AzureDevopsClient.java:131]: To be done as a separate task.

File: src/test/java/com/checkmarx/flow/cucumber/component/webhook/WebHookSteps.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (AvivCX) [reply src/test/java/com/checkmarx/flow/cucumber/component/webhook/WebHookSteps.java:49]: why the automation infra doesn't run runCxFlowAsService?

[the remark this answers is not in the document]
Comment (alex-ko-dev) [reply src/test/java/com/checkmarx/flow/cucumber/component/webhook/WebHookSteps.java:49]: I don't see a reason why it should. Most tests don't require it. Starting the web app for each test will just make the tests run longer.

**Pull request merged**